File: src/main/scala/com/xebia/functional/llm/openai/models/CompletionChoice.scala
package com.xebia.functional.llm.openai.models

import com.theokanning.openai.completion.{CompletionChoice => JCompletionChoice}

final case class CompletionChoice(
    text: String,
    index: Integer,
    finishReason: String
)

object CompletionChoice:
  def fromJava(j: JCompletionChoice): CompletionChoice =
    CompletionChoice(j.getText(), j.getIndex(), j.getFinish_reason())

File: src/jvmMain/kotlin/llm/models/CompletionChoice.kt
package llm.models

import arrow.core.raise.nullable
import com.theokanning.openai.completion.CompletionChoice as JCompletionChoice

fun CompletionChoice.Companion.fromJava(j: JCompletionChoice): CompletionChoice? = nullable {
  val text = j.text.bind()
  val index = j.index.bind()
  val finishReason = j.finish_reason.bind()
  CompletionChoice(text, index, finishReason)
}
